src/gfnff/gdisp0.f90

```diff
@@ -130,6 +130,7 @@ subroutine weight_references_d4(dispm, nat, atoms, wf, cn, gwvec, gwdcn)
          norm = norm + gw
          dnorm = dnorm + 2*wf*(dispm%cn(iref, ati) - cn(iat)) * gw
       end do
+      if (norm.eq.0.0_wp) cycle
```

Comment:
What does this `cycle` lead to? Why has it not been required before?

Reply:
It's something I found while debugging with valgrind. I would expect that it has been required before, but the compiler and compiler options we use already seem to handle problems with division by zero. However, we should not divide by zero, that is why I added the cycle to continue with the next atom if the norm is zero for the current atom. As checked by the unit tests, this does not influence energy, gradient, or sigma.